owncloud/web
# [full-ci][tests-only] run acceptance tests using files for upload from the middleware

### Description:
use `filesForUpload` from the middleware repo to run web acceptance tests

### Related:
- see also the description in https://github.com/owncloud/owncloud-test-middleware/pull/71
- blocker in https://github.com/owncloud/web/pull/6188

### Motivation:
- prepare files/folders to upload for upload tests (with any client) from the middleware, so that we can assert the downloaded content with the local content easily.
- If we want to test an upload feature using a locally created file, then:
  - web-test have to create a file locally
  - save it
  - upload it to the selenium server
  - upload it to owncloud using the webUI
  - download the file using the webUI
  - downloaded content should match the local file content

- now the same tests may be needed for the desktop client or any other client. we use middleware to solve duplication issues and move steps like:
  - creating a local file
  - asserting downloaded content with local file
to the middleware so that every client can take benefit from this.

- also in the process, the folder used by middleware to manage filesForUploads can be shared with selenium (thanks to docker), and now we even do not have to upload these files to the selenium (since they are already synced from middleware dir.). then, we can make an upload test with ease.
